src/plugins/liveobjects/liveobjects.ts

```diff
@@ -20,14 +28,24 @@ export class LiveObjects {
   constructor(channel: RealtimeChannel) {
     this._channel = channel;
     this._client = channel.client;
+    this._eventEmitter = new this._client.EventEmitter(this._client.logger);
     this._liveObjectsPool = new LiveObjectsPool(this);
     this._syncLiveObjectsDataPool = new SyncLiveObjectsDataPool(this);
     this._syncInProgress = true;
   }
 
   async getRoot(): Promise<LiveMap> {
-    // TODO: wait for SYNC sequence to finish to return root
-    return this._liveObjectsPool.get(ROOT_OBJECT_ID) as LiveMap;
+    if (!this._syncInProgress) {
+      // SYNC is finished, can return immediately root object from pool
+      return this._liveObjectsPool.get(ROOT_OBJECT_ID) as LiveMap;
+    }
+
+    // otherwise wait for SYNC sequence to finish
+    return new Promise((res) => {
+      this._eventEmitter.once(LiveObjectsEvents.SyncCompleted, () => {
+        res(this._liveObjectsPool.get(ROOT_OBJECT_ID) as LiveMap);
+      });
+    });
```

Comment:
```suggestion
    // SYNC is currently in progress, wait for SYNC sequence to finish
    if (this._syncInProgress) {
      await this._eventEmitter.once(LiveObjectsEvents.SyncCompleted)
    }
    
    return this._liveObjectsPool.get(ROOT_OBJECT_ID) as LiveMap;
```

^ much simpler + avoids mixing promises + async/await

Reply:
oh cool, didn't realize that there is an `.once` overload that returns a Promise. accepted the change